hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/validators/ConsensusCustomFeesValidator.java

```diff
@@ -0,0 +1,134 @@
+import java.util.List;
+import javax.inject.Inject;
+
+public class ConsensusCustomFeesValidator {
+    /**
+     * Constructs a {@link ConsensusCustomFeesValidator} instance.
+     */
+    @Inject
+    public ConsensusCustomFeesValidator() {
+        // Needed for Dagger injection
+    }
+
+    /**
+     * Validates custom fees for {@code ConsensusCreateTopic} operation.
+     *
+     * @param accountStore       The account store.
+     * @param tokenRelationStore The token relation store.
+     * @param tokenStore         The token store.
+     * @param customFees         The custom fees to validate.
+     * @param expiryValidator    The expiry validator to use (for fee collector accounts)
+     */
+    public void validateForCreation(
+            @NonNull final ReadableAccountStore accountStore,
+            @NonNull final ReadableTokenRelationStore tokenRelationStore,
+            @NonNull final ReadableTokenStore tokenStore,
+            @NonNull final List<ConsensusCustomFee> customFees,
+            @NonNull final ExpiryValidator expiryValidator) {
+        requireNonNull(accountStore);
+        requireNonNull(tokenRelationStore);
+        requireNonNull(tokenStore);
+        requireNonNull(customFees);
+        requireNonNull(expiryValidator);
+
+        for (final var fee : customFees) {
+            // Validate the fee collector account is in a usable state
+            getIfUsableForAliasedId(
+                    fee.feeCollectorAccountIdOrElse(AccountID.DEFAULT),
+                    accountStore,
+                    expiryValidator,
+                    INVALID_CUSTOM_FEE_COLLECTOR);
+
+            final var isSpecified = fee.hasFixedFee();
+            validateTrue(isSpecified, CUSTOM_FEE_NOT_FULLY_SPECIFIED);
+            validateFixedFeeForCreation(fee, tokenRelationStore, tokenStore);
+        }
+    }
+
+    private void validateFixedFeeForCreation(
+            @NonNull final ConsensusCustomFee fee,
+            @NonNull final ReadableTokenRelationStore tokenRelationStore,
+            @NonNull final ReadableTokenStore tokenStore) {
+        final var fixedFee = fee.fixedFeeOrThrow();
+        validateTrue(fixedFee.amount() > 0, CUSTOM_FEE_MUST_BE_POSITIVE);
+        if (fixedFee.hasDenominatingTokenId()) {
+            validateExplicitTokenDenomination(
+                    fee.feeCollectorAccountId(),
+                    fixedFee.denominatingTokenId(),
+                    fixedFee.amount(),
+                    tokenRelationStore,
+                    tokenStore);
+        }
+    }
+
+    /**
+     * Validate explicitly set token denomination for custom fees.
+     *
+     * @param feeCollectorNum The fee collector account number.
+     * @param tokenNum The token number used for token denomination.
+     * @param feeAmount The fee amount.
+     * @param tokenRelationStore The token relation store.
+     * @param tokenStore The token store.
+     */
+    private void validateExplicitTokenDenomination(
```

Comment:
The first thought I got was if we have a way to extract this common logic to a base class. But, since they are in different services, I think its not possible ? Just adding as a thought 😄

Reply:
Can we add this validator just like AttributeValidator in SPI ? Not sure. We can do the refactor later if you want. @tinker-michaelj @netopyr  thoughts ?